### DISCUSSION TITLE
Import errors

### DISCUSSION THREAD

**Discussion by testfailagain** [microsoft/pyright#3158] [Q&A]
Hi,

It's not so clear to me how should I configure pyrightconfig.json file.
Hi have a proyect with this structure:
level0/**venv**
----/level1/
-------------level2.1/
------------------------djangoapp
------------- level2.2/
-------------------------- djangoapp
-------------level2.3/
-------------------------- djangoapp

And I reach this point, in each django project (but I think I could have only one file):
{
  "include" : ["/home/bla/level0/level1/level2.1"],
  "venvPath": "/home/bla/level0",
  "venv": /venv",
  "executionEnvironments": [
    {
      "root": "/home/bla/level0/level1/level2.1/src"
     }
    ]
  }
  
  (Ignore typo errors, I copy it by hand)
  
  So, what am I doing wrong?
  Can I have one file to rule them all? (like I suppouse)
  There's another guide more specific (and more simple for noobs)

**Comment by erictraut** [comment]
Converting to a discussion since this isn't a bug.

**Reply by erictraut** [reply]
For configuration documentation and sample config files, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/configuration.md).

A couple of pointers...
* The "include" and "root" paths are intended to be relative to the root of your project, but you are using absolute paths here.
* It's not generally recommended that you include the "venvPath" in your config file. The config file is something that is generally checked in to the repo, and each developer will likely have a venvPath in different locations. If you have standardized on a venv within the project directory, then I guess it's OK to include it here, but the path should be relative (to the project root), not absolute.
* The "venv" configuration should refer to the name of a venv, not a path.

**Reply by testfailagain** [reply]
The documentation is the same that I tried to follow and is not clear (for me).
I update the file:
level0/pyrightconfig.json
```
{
    "include" : ["."],
    "venvPath": ".",
    "venv": "venv",
    "executionEnvironments": [
        {
            "root": "level1/level2.1/src"
        }
    ]
}
```
but still is not working.

**Reply by erictraut** [reply]
With this structure, you shouldn't include "." because that includes your entire venv, and you don't want to include all of the files in your venv as part of your project. So you should change it to `"include": ["level1"]`. The rest looks OK to me.

**Reply by testfailagain** [reply]
nothing, I'll try more later.
Meanwhile, how can I add other projects paths? like level2.2 be recognize by level2.1

**Comment by testfailagain** [comment]
I see my error, I don't have django installed in the venv, only in docker, where I execute the code and works.
So now, I only have one questión ¿how can I point to other code? (no installable)
